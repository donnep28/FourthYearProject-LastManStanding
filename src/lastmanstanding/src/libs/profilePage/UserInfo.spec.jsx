import React from 'react';
import {shallow} from 'enzyme';
import UserInfo from './userInfo';

describe('UserInfo', () => {
  beforeAll(() => {
    jest.resetAllMocks();
  });

  const props = {
    myInfo: [
      {
        "Sub": "ecbc56df-d81a-410d-b399-4421b55ced32",
        "Username": "philip",
        "email": "philip.donnelly28@mail.dcu.ie",
        "favouriteTeam": "manchesterunited",
        "leagueIDs": [
          "Plips League#283625",
          "Glenn's Last Man Standing#522084",
          "Ladyboys Inc#819148",
          "The Last man standing#984785",
        ],
        "losses": "5",
        "wins": "0",
      }
    ],
    setmyTeam: jest.fn(),
  };

  describe('Snapshot', () => {
    it('renders correctly', () => {
      const component = shallow(
          <UserInfo {...props}/>,
      );
      expect(component).toMatchSnapshot();
    });
  });
  describe('change team', () => {
    it('sets setmyTeam to true', () => {
      const component = shallow(
          <UserInfo {...props}/>,
      );
      component.find("WithStyles(ForwardRef(Button))").simulate("click");
      expect(props.setmyTeam).toHaveBeenCalledWith(true);
    });
  });
});