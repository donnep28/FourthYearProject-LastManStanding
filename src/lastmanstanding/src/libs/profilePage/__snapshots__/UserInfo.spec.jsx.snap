// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo Snapshot renders correctly 1`] = `
<Fragment>
  <WithStyles(ForwardRef(ListItem))>
    Username: 
    philip
  </WithStyles(ForwardRef(ListItem))>
  <WithStyles(ForwardRef(ListItem))>
    Email: 
    philip.donnelly28@mail.dcu.ie
  </WithStyles(ForwardRef(ListItem))>
  <WithStyles(ForwardRef(ListItem))>
    Favourite Team: 
    Manchester United FC
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
      style={
        Object {
          "minWidth": "36px",
        }
      }
    >
      <Memo(ForwardRef(AddCircleOutlineIcon)) />
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(ListItem))>
</Fragment>
`;
